text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/internal/parser/TokenParser.java

```diff
@@ -246,8 +246,11 @@ public static void parseString(final String message, final MatchedTokenConsumer<
               break;
             case '\'':
             case '"':
-              state = FirstPassState.STRING;
               currentStringChar = (char) codePoint;
+              // Look ahead if the quote being opened is ever closed
+              if (message.substring(i + 1).indexOf(codePoint) != -1) {
```

Comment:
```suggestion
              if (message.indexOf(codePoint, i + 1) != -1) {
```

just to save that allocation :p